internal/pubsub/pubsub.go

```diff
@@ -96,6 +103,64 @@ func (ps *PubSub[T, M]) Subscribe(mask M, subCh chan T, doneCh <-chan struct{},
 	return nil
 }
 
+// SubscribeJSON - Adds a subscriber to pubsub system and returns results with JSON encoding.
+func (ps *PubSub[T, M]) SubscribeJSON(mask M, subCh chan<- []byte, doneCh <-chan struct{}, filter func(entry T) bool) error {
+	totalSubs := atomic.AddInt32(&ps.numSubscribers, 1)
+	if ps.maxSubscribers > 0 && totalSubs > ps.maxSubscribers {
+		atomic.AddInt32(&ps.numSubscribers, -1)
+		return fmt.Errorf("the limit of `%d` subscribers is reached", ps.maxSubscribers)
+	}
+	ps.Lock()
+	defer ps.Unlock()
+	subChT := make(chan T, 10000)
+	sub := &Sub[T]{ch: subChT, types: Mask(mask.Mask()), filter: filter}
+	ps.subs = append(ps.subs, sub)
+
+	// We hold a lock, so we are safe to update
+	combined := Mask(atomic.LoadUint64(&ps.types))
+	combined.Merge(Mask(mask.Mask()))
+	atomic.StoreUint64(&ps.types, uint64(combined))
+	var encErr atomic.Pointer[error]
```

Comment:
We are keeping track of the error here, but subsequently not doing anything with it (not that I would know what to do with it...).

So we may just as well ignore it (or comment in the code that we're not doing something purposeful with it)

Reply:
@klauspost ^^